app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,29 @@
+class ReportsController < ApplicationController
+  before_action :redirect_login
+
+  def show; end
+
+  def new
+    return unless logged_in?
```

Comment:
cần dòng này nữa k

Reply:
Không cần nữa ạ. Tại vì `before_action :redirect_login` nó đã kiểm tra logged_in? trước rồi ạ. Nếu chưa log in thì chuyển hướng về trang log in.